doc/ONBOARDING.md

```diff
@@ -0,0 +1,102 @@
+# Temporal Server Local Setup Guide
```

Comment:
What is it for? We recomend to use `docker-compose` for local development. Is there some specific scenario?

Reply:
Also `doc` dir was intentionally removed from this repo recently.

Reply:
i can change the dir name.

> We recomend to use docker-compose for local development
this may not be applicable to some users, due to perf reason or apple M1

Reply:
I think it should go to subdir under `/.development` dir. I can't figure out the best name but something like `host`, or `local`, ot `dockerless`, or something else. And let file name be just `README.md` (opened by default in the dir by github).

Also it would be great to add these foreword about apple M1. Like when and why to use it instead of generic `docker-compose` approach.